llama_stack/providers/inline/agents/meta_reference/agent_instance.py

```diff
@@ -246,6 +247,129 @@ async def create_and_execute_turn(self, request: AgentTurnCreateRequest) -> Asyn
 
             yield chunk
 
+    async def resume_turn(self, request: AgentTurnResumeRequest) -> AsyncGenerator:
+        with tracing.span("resume_turn") as span:
+            span.set_attribute("agent_id", self.agent_id)
+            span.set_attribute("session_id", request.session_id)
+            span.set_attribute("turn_id", request.turn_id)
+            span.set_attribute("request", request.model_dump_json())
+            assert request.stream is True, "Non-streaming not supported"
+
+            session_info = await self.storage.get_session_info(request.session_id)
+            if session_info is None:
+                raise ValueError(f"Session {request.session_id} not found")
+
+            turns = await self.storage.get_session_turns(request.session_id)
+
+            messages = []
+            if self.agent_config.instructions != "":
+                messages.append(SystemMessage(content=self.agent_config.instructions))
+
+            for i, turn in enumerate(turns):
+                messages.extend(self.turn_to_messages(turn))
```

Comment:
this is important code which is duplicated, maybe there should be a small `def turn_history_to_messages(session_id)` refactored method used from both create_turn and resume_turn

Reply:
done!